src/plugins/ds_mongo.c

```diff
@@ -0,0 +1,3400 @@
+        goto cleanup;
+    }
+
+cleanup:
+    bson_destroy(doc);
+    return err_info;
+}
+
+/**
+ * @brief Create candidate-modified flag.
+ * 
+ * @param[in] module Given MongoDB collection.
+ * @param[in] modified Whether candidate datastore is modified.
+ * @return NULL on success;
+ * @return Sysrepo error info on error.
+ */
+static sr_error_info_t *
+srpds_insert_candidate_modified_flag(mongoc_collection_t *module, int modified)
+{
+    sr_error_info_t *err_info = NULL;
+    bson_error_t error;
+    bson_t *doc = NULL;
+
+    doc = BCON_NEW("_id", BCON_UTF8("1"), "modified", BCON_BOOL(modified));
+    if (!mongoc_collection_insert_one(module, doc, NULL, NULL, &error)) {
+        ERRINFO(&err_info, SR_ERR_OPERATION_FAILED, "mongoc_collection_insert_one()", error.message)
+        goto cleanup;
+    }
+
+cleanup:
+    bson_destroy(doc);
+    return err_info;
+}
+
+/**
+ * @brief Load the whole data tree (only for operational).
+ * 
+ * @param[in] mod_data Whole data tree.
+ * @param[out] diff_data Helper structure for storing diff operations.
+ * @return NULL on success;
+ * @return Sysrepo error info on error.
+ */
+static sr_error_info_t *
+srpds_load_oper_recursively(const struct lyd_node *mod_data, struct mongo_diff_data *diff_data)
+{
+    sr_error_info_t *err_info = NULL;
+    const struct lyd_node *sibling = mod_data;
+    struct lyd_node *child = NULL;
+    struct lyd_meta *meta = NULL;
+    struct lys_module *module = NULL; // for opaque nodes
+    struct lyd_attr *attr = NULL; // for opaque nodes
+    const char *meta_value, *predicate = NULL;
+    char *path_no_pred = NULL;
+    char *path = NULL, *final_path = NULL;
+    const char *value = NULL;
+    char *any_value = NULL;
+    int32_t valtype;
+    bson_t *bson_query = NULL;
+    
+    while (sibling) {
+        /* get path */
+        if ((err_info = srpds_get_predicate(sibling, &predicate, &path, &path_no_pred))) {
+            goto cleanup;
+        }
+
+        /* LYD_ANYDATA_XML */
+        valtype = 0;
+
+        /* get value */
+        if (sibling->schema && (sibling->schema->nodetype & LYD_NODE_ANY)) {
+            /* these are JSON data, set valtype */
+            if (((struct lyd_node_any *)sibling)->value_type == LYD_ANYDATA_JSON) {
+                /* LYD_ANYDATA_JSON */
+                valtype = 1;
+            }
+
+            /* lyd_node_any */
+            if (lyd_any_value_str(sibling, &any_value) != LY_SUCCESS) {
+                ERRINFO(&err_info, SR_ERR_LY, "lyd_any_value_str()", "")
+                goto cleanup;
+            }
+            value = any_value;
+        }
+        else {
+            /* node values, opaque node values and rest */
+            value = lyd_get_value(sibling);
+        }
+
+        /* create all data,
+        there is no need to store order for userordered lists and leaflists
+        since MongoDB does load all data in the same order in which they were stored */
```

Comment:
Multiline comments should always look like:
```
/* text, more text
 * some more text,
 * and end. */
```

Reply:
done